src/types/callable/function.js

```diff
@@ -12,14 +12,14 @@ export function createFunction(info) {
     const inArgs = parseInArgs(...args);
     const outArgs = initOutArgs();
 
-    const error = new ArrayBuffer(16);
-    const returnValue = new ArrayBuffer(8);
+    const error = new ArrayBuffer(8);
+    const returnValue = new BigUint64Array(1);
 
     const success = g.function_info.invoke(
       info,
       new BigUint64Array(inArgs),
```

Comment:
maybe `parseInArgs` should also return a `BigUint64Array` for consistency?

Reply:
in instance methods, you need to add self argument later, so it's easier to use dynamic array.